### PR TITLE
kubevirt CR: GA for the free page reporting feature

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: enp0s3 <ibezukh@redhat.com>

**What this PR does / why we need it**:
Kubevirt FPR feature was graduated to GA, therefore enable it in the Kubevirt CR

**Reviewer Checklist**

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:
```jira-ticket
https://issues.redhat.com/browse/CNV-29041
```

**Release note**:
```release-note
kubevirt CR: GA for the free page reporting feature
```

### CODE REVIEW

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -205,8 +205,6 @@ type HyperConvergedSpec struct {
 	VMStateStorageClass *string `json:"vmStateStorageClass,omitempty"`
 
 	// VirtualMachineOptions holds the cluster level information regarding the virtual machine.
-	// +kubebuilder:default={"disableFreePageReporting": true}
-	// +default={"disableFreePageReporting": true}
```
Comment (tiraboschi) [inline api/v1beta1/hyperconverged_types.go:209]: I'm not so sure that this will be enough for who initially deployed with the previous release and now upgrades.
What are we supposed to do in that case?

Comment (enp0s3) [reply api/v1beta1/hyperconverged_types.go:209]: @tiraboschi Thank you for the review! I would expect that during operator rollout, any change in defaults will take effect, unless the field was modified by the user. Does HCO have such mechanism?

File: assets/upgradePatches.json [hunk 2]
```diff
@@ -37,6 +37,16 @@
           }
         }
       ]
+    },
+    {
+      "semverRange": "<=1.10.0",
```
Comment (tiraboschi) [inline assets/upgradePatches.json:42]: "semverRange": "<1.11.0",

File: assets/upgradePatches.json [hunk 3]
```diff
@@ -37,6 +37,16 @@
...
+      "jsonPatch": [
+        {
+          "op": "replace",
+          "path": "/spec/virtualMachineOptions/disableFreePageReporting",
+          "value": false
```
Comment (tiraboschi) [inline assets/upgradePatches.json:47]: "op": "remove",
          "path": "/spec/virtualMachineOptions/disableFreePageReporting"

File: tests/func-tests/defaults_test.go [hunk 4]
```diff
@@ -212,7 +212,7 @@ var _ = Describe("Check Default values", Label("defaults"), Serial, func() {
 
 	Context("virtualMachineOptions defaults", func() {
```
Comment (nunnatsa) [inline tests/func-tests/defaults_test.go:213]: The defaults test fails now. Please remove this test (the whole `Context("virtualMachineOptions defaults")`)

There is no point to check it. If you like, you can add a test to make sure that `virtualMachineOptions` is `nil`.

File: assets/upgradePatches.json [hunk 5]
```diff
@@ -37,6 +37,25 @@
...
+        {
+          "op": "test",
+          "path": "/spec/virtualMachineOptions",
+          "value": "[*]"
+        },
+        {
+          "op": "test",
+          "path": "/spec/virtualMachineOptions/disableFreePageReporting",
+          "value": "true"
+        },
```
Comment (tiraboschi) [inline assets/upgradePatches.json:53]: see allowMissingPathOnRemove from https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2535

File: tests/func-tests/defaults_test.go (hunk 4 above)
Comment (enp0s3) [reply tests/func-tests/defaults_test.go]: Done.

**Comment by tiraboschi** [comment]
> @nunnatsa @tiraboschi I have also added GH actions configuration to skip the jobs in case of a draft PR. If you want I can drop it or do it in a separate PR.

yes, please

**Comment by enp0s3** [comment]
@tiraboschi @nunnatsa Could you please have another look?

**Review by tiraboschi — APPROVED** [review]
